### PR TITLE
basic inverse API

### PR DESCRIPTION
**Pull request by rijobro**
Basic inverse transform API.

### Description
Append to list of invertible transform, perform inverse with pop from list (LIFO).

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/inverse.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            whether `do_transform` was True or False and can only contain objects that are
+            accepted in pytorch's batch (e.g., `None` is not allowed).
+        4. Implement an `inverse` method. Make sure that after performing the inverse,
+            `remove_most_recent_transform` is called.
+    """
+
+    class Keys(Enum):
+        """Extra meta data keys used for inverse transforms."""
+
+        class_name = "class"
```
Comment (Nic-Ma) [inline monai/transforms/inverse.py:67]: I think it's better to use capital words for the keys, to be consistent with all the existing MONAI enums:
https://github.com/Project-MONAI/MONAI/blob/master/monai/utils/enums.py

**Review by Nic-Ma — APPROVED** [review]
Looks good to me except for a minor comment.

Thanks.

File: monai/transforms/inverse.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        #. Any extra information that might be needed for the inverse can be included with the
+           dictionary ``extra_info``. This dictionary should have the same keys regardless of
+           whether ``do_transform`` was `True` or `False` and can only contain objects that are
+           accepted in pytorch data loader's collate function (e.g., `None` is not allowed).
+        #. Implement an ``inverse`` method. Make sure that after performing the inverse,
+           ``pop_transform`` is called.
+
+    """
+
+    class Keys(Enum):
```
[the remark this answers is not in the document]
Comment (rijobro) [reply monai/transforms/inverse.py]: done

File: monai/transforms/inverse.py (hunk 1 above)
Comment (rijobro) [reply monai/transforms/inverse.py]: done

File: tests/test_inverse.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    LoadImaged,
+    ResizeWithPadOrCrop,
+    SpatialPadd,
+    allow_missing_keys_mode,
+)
+from monai.utils import first, optional_import, set_determinism
+from tests.utils import make_nifti_image, make_rand_affine
+
+if TYPE_CHECKING:
+    import matplotlib.pyplot as plt
```
[the remark this answers is not in the document]
Comment (rijobro) [reply tests/test_inverse.py]: done

File: monai/transforms/inverse.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    def get_most_recent_transform(self, data: dict, key: Hashable) -> dict:
+        """Get most recent transform."""
+        transform = dict(data[str(key) + InverseKeys.KEY_SUFFIX.value][-1])
+        self.check_transforms_match(transform)
+        return transform
+
+    def pop_transform(self, data: dict, key: Hashable) -> None:
+        """Remove most recent transform."""
+        data[str(key) + InverseKeys.KEY_SUFFIX.value].pop()
```
Comment (wyli) [inline monai/transforms/inverse.py:103]: had another look at this class, perhaps `get_most_recent_transform` could be merged into `pop_transform`?

```suggestion
    def pop_transform(self, data: dict, key: Hashable) -> None:
        """Removes and returns most recent transform."""
        xform = dict(data[str(key) + InverseKeys.KEY_SUFFIX.value].pop())
        self.check_transforms_match(xform)
        return xform
```

Comment (rijobro) [reply monai/transforms/inverse.py:103]: I don't really want to pop when getting the most recent transform as I only want to remove it from the list once the inverse has been successfully applied.

Comment (wyli) [reply monai/transforms/inverse.py:103]: ok, at the moment I find the methods are always used together... it seems it's safe to merge them, or do you mean do you want to recover from some inversion failure in the future?

**Pull request merged**